lib/authorizations.js

```diff
@@ -0,0 +1,35 @@
+/*!
+ * Copyright (c) 2019-2023 Digital Bazaar, Inc. All rights reserved.
+ */
+import * as bedrock from '@bedrock/core';
+import forwarded from 'forwarded';
+import {verify} from '@bedrock/turnstile';
+
+const {config, util: {BedrockError}} = bedrock;
+
+export async function authorizeRegistration({req}) {
+  const cfg = config['account-http'];
+  const {authorizationRequired} = cfg.registration;
+  const {authorization} = req.body;
+  if(!(authorization?.type === authorizationRequired)) {
+    throw new BedrockError(
+      `Authorization using "${authorizationRequired}" is required.`, {
+        name: 'NotAllowedError',
+        details: {
+          httpStatusCode: 403,
+          public: true
+        }
+      });
+  }
+
+  // the first IP in the sourceAddresses array will *always* be the IP
+  // reported by Express.js via `req.connection.remoteAddress`. Any additional
+  // IPs will be from the `x-forwarded-for` header.
+  const sourceAddresses = forwarded(req);
+
+  // to get the end users IP adddress, use first IP from the `x-forwarded-for`
+  // header
+  const remoteIp = sourceAddresses[1];
```

Comment:
Does this work in dev mode? Should it? Does this need to be more intelligent to figure this out?

Reply:
`remoteIp` is not required by Turnstile, so in the case of dev mode it is undefined and the request is still able to process. From reading the docs, this seemed to be the best/easiest way to get the end user's IP address. Is there a better way to do it?

Reply:
I guess we'll move forward with this and if it's problematic make adjustments. Thanks!